pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -163,53 +164,61 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 			reqLogger.Error(err1, "Could not update error conditions")
 		}
 		return *recResult, err
-	} else {
-		if infinispan.RemoveCondition("preliminaryChecksFailed") {
-			err1 := r.client.Status().Update(context.TODO(), infinispan)
-			reqLogger.Error(err1, "Could not update error conditions")
-		}
 	}
+	if infinispan.RemoveCondition("preliminaryChecksFailed") {
+		err1 := r.client.Status().Update(context.TODO(), infinispan)
+		reqLogger.Error(err1, "Could not update error conditions")
+	}
+
+	infinispan.ApplyEndpointEncryptionSettings(kubernetes.GetServingCertsMode(), reqLogger)
 
-	// Check x-site configuration first.
-	// Must be done before creating any Infinispan resources,
-	// because remote site host:port combinations need to be injected into Infinispan.
 	xsite := &configuration.XSite{}
-	err = xsite.ComputeXSite(infinispan, kubernetes, r.scheme, reqLogger)
+	if infinispan.HasSites() {
+
+		// Check x-site configuration first.
+		// Must be done before creating any Infinispan resources,
+		// because remote site host:port combinations need to be injected into Infinispan.
+
+		// For cross site, reconcile must come before compute, because
+		// we need xsite service details to compute xsite struct
+		siteService, err := r.reconcileXSite(infinispan, r.scheme, reqLogger)
+		if err != nil {
+			reqLogger.Error(err, "Error in reconcileXSite", "Infinispan.Namespace", infinispan.Namespace)
+			return reconcile.Result{}, err
+		}
+
+		err = xsite.ComputeXSite(infinispan, kubernetes, siteService, reqLogger)
+		if err != nil {
+			reqLogger.Error(err, "Error in computeXSite", "Infinispan.Namespace", infinispan.Namespace)
+			return reconcile.Result{}, err
+		}
+	}
+
+	configMap, err := r.computeConfigMap(xsite, infinispan)
+	if err != nil {
+		reqLogger.Error(err, "could not create Infinispan configuration")
+		return reconcile.Result{}, err
+	}
+
+	err = r.reconcileConfigMap(infinispan, configMap, reqLogger)
 	if err != nil {
-		reqLogger.Error(err, "Error in computeXSite", "Infinispan.Namespace", infinispan.Namespace)
+		reqLogger.Error(err, "Error in reconcileConfigMap", "Infinispan.Namespace", infinispan.Namespace)
```

Comment:
```suggestion
		reqLogger.Error(err, "Error in reconcileConfigMap")
```
Looks like we already have an "Infinispan.Namespace" in the root logger